components/Editor/index.tsx

```diff
@@ -82,49 +93,94 @@ const Editor = ({ readOnly = false }: Props) => {
   const [callValue, setCallValue] = useState('')
   const [unit, setUnit] = useState(ValueUnit.Wei as string)
 
-  const handleWorkerMessage = (event: MessageEvent) => {
-    const { code: byteCode, warning, error } = event.data
+  const log = useCallback(
+    (line: string, type = 'info') => {
+      // See https://blog.logrocket.com/a-guide-to-usestate-in-react-ecb9952e406c/
+      setOutput((previous) => {
+        const cloned = previous.map((x) => ({ ...x }))
+        cloned.push({ type, message: line })
+        return cloned
```

Comment:
you could use an array spreading here instead of a func, smth like:

```
setOutput([...output, { type, message: line }])
```

Reply:
In the link they say you have to make sure to do a deep copy, including the object in the array, and as far as I understood your example would not do that

Reply:
Oh, you're right. We're having an array of objects, and not primitives here. That's cool then